### PR TITLE
migration from gradle to maven

### PR DESCRIPTION
**Pull request by zjaishu**
Migrated from gradle to maven.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,4 +11,5 @@ build/
 
 .classpath
 .project
-.settings/
\ No newline at end of file
+.settings/
+/target/
```
Comment (SantoshNC68) [inline .gitignore:7]: Is this file required ??

File: index.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# PRI 
+The versions are indicated by a version identification of N.0.X. This shall be read as the actually released version of the Artifact in the ARM is the version N.0 with the highest X. Once a release has been made, N is stepped to N+1.
+
+## 0.0.X
```
Comment (SantoshNC68) [inline index.md]: check the version and update it properly here

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	<build>
+		<plugins>
+			<plugin>
+				<artifactId>maven-compiler-plugin</artifactId>
+				<version>3.1</version>
+				<configuration>
+					<source>1.8</source>
+					<target>1.8</target>
+					<fork>true</fork>
+					<executable>C:\Program Files\Java\jdk1.8.0_151\bin\javac</executable>
```
Comment (SantoshNC68) [inline pom.xml]: remove the executable dependecy of local file system. Is it required at all ??

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
```
Comment (SantoshNC68) [inline pom.xml]: remove tabs and replace with 4 spaces in this file

**Review by SantoshNC68 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (zjaishu) [reply .gitignore:7]: yes we are ignoring these folder while pushing to git from local

File: index.md (hunk 2 above)
Comment (xdurvak) [reply index.md]: there is no need of this file you can remove it, anyway we are maintaining changelog.md file

Comment (zjaishu) [reply index.md]: okie

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
+            <version>2.8.1</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>commons-codec</groupId>
+            <artifactId>commons-codec</artifactId>
+            <version>1.10</version>
```
Comment (raja-maragani) [inline pom.xml]: Please use global properties for version declaration

**Review by raja-maragani — CHANGES_REQUESTED** [review]

**Review by raja-maragani — APPROVED** [review]

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
+            <version>${google-gson.version}</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>commons-codec</groupId>
```
Comment (raja-maragani) [inline pom.xml]: Please check, are we using this functionality? if not please remove this

**Review by raja-maragani — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 6 above)
Comment (zjaishu) [reply pom.xml]: yes we are not using these dependency. we can remove

File: pom.xml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+    xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>com.github.Ericsson</groupId>
+    <artifactId>eiffel-remrem-shared</artifactId>
+    <version>0.3.7</version>
+    <properties>
+        <junit.version>4.12</junit.version>
+        <google-gson.version>2.8.1</google-gson.version>
+        <commons-codec.version>1.10</commons-codec.version>
```
Comment (raja-maragani) [inline pom.xml]: remove unused version declaration.

**Review by raja-maragani — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+                <version>3.1</version>
+                <configuration>
+                    <source>1.8</source>
+                    <target>1.8</target>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+    <dependencies>
+        <dependency>
```
Comment (raja-maragani) [inline pom.xml]: This one also we can remove, 
Only one class we are using JsonParser from this library, in VersionService.java. 
and it is a just Object declaration, so please remove

**Review by raja-maragani — CHANGES_REQUESTED** [review]

**Review by SantoshNC68 — APPROVED** [review]

File: pom.xml (hunk 8 above)
Comment (zjaishu) [reply pom.xml]: okie

File: pom.xml (hunk 7 above)
Comment (zjaishu) [reply pom.xml]: okie

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
+        </dependency>
+    </dependencies>
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.1</version>
```
Comment (raja-maragani) [inline pom.xml]: try 3.6.1 version

**Review by raja-maragani — APPROVED** [review]
